NPE in maven/ibm-javaee8-webprofile-liberty quickstart

### Description

When running the `quickstarts/maven/ibm-javaee8-webprofile-liberty` example a NPE is thrown

```bash
$ mvn -P'integration-tests' clean verify -DtestProxyHost=$(minikube ip)
```

```bash
Caused by: java.lang.NullPointerException
    at org.eclipse.jkube.kit.build.service.docker.BuildService.autoPullBaseImage (BuildService.java:250)
    at org.eclipse.jkube.kit.build.service.docker.BuildService.buildImage (BuildService.java:75)
    at org.eclipse.jkube.kit.config.service.kubernetes.DockerBuildService.buildSingleImage (DockerBuildService.java:59)
    at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.processImage (AbstractImageBuildService.java:53)
    at org.eclipse.jkube.kit.config.service.AbstractImageBuildService.build (AbstractImageBuildService.java:34)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.buildAndTag (AbstractDockerMojo.java:631)
...

```

This is caused during the build phase because it's not being skipped. The behavior started in v1.6.0, in v1.5.1 this wasn't a problem because `ImageConfiguration.buildConfig` was checked before proceeding with the build:

https://github.com/eclipse/jkube/blob/3455471f7492508ce6ec220642b31d8339e3b8fd/kubernetes-maven-plugin/plugin/src/main/java/org/eclipse/jkube/maven/plugin/mojo/build/AbstractDockerMojo.java#L487-L493

~~I'm still not sure if we actually removed the check for the buildconfig to allow other scenarios.~~

The check seems to be here, but is incomplete:

https://github.com/eclipse/jkube/blob/3f17790e0e34dcbbb8b9f8209f743701514104e8/jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/AbstractImageBuildService.java#L47-L57


### Related to
- https://github.com/eclipse/jkube/issues/1197
- https://github.com/eclipse/jkube/pull/1224

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ Usage:
 * Fix #1468: Add startup probe in QuarkusHealthCheckEnricher
 * Fix #1473: Add OpenLibertyHealthCheckEnricher to automatically add health checks in OpenLiberty projects
 * Fix #1537: Registry not set up in `oc:build`
+* Fix #1654: images with missing build configuration should not be built
 
 
 ### 1.8.0 (2022-05-24)

--- a/jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/AbstractImageBuildService.java
+++ b/jkube-kit/config/service/src/main/java/org/eclipse/jkube/kit/config/service/AbstractImageBuildService.java
@@ -49,6 +49,8 @@ public abstract class AbstractImageBuildService implements BuildService {
       for (ImageConfiguration imageConfiguration : imageConfigurations) {
         if (imageConfiguration.getBuildConfiguration() != null && imageConfiguration.getBuildConfiguration().getSkip()) {
           jKubeServiceHub.getLog().info("%s : %s", imageConfiguration.getDescription(), skipMessage);
+        } else if (imageConfiguration.getBuildConfiguration() == null) {
+          jKubeServiceHub.getLog().info("%s : %s (Image configuration has no build settings)", imageConfiguration.getDescription(), skipMessage);
         } else {
           imageConfigurationConsumer.process(imageConfiguration);
         }

--- a/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/DockerBuildServiceTest.java
+++ b/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/DockerBuildServiceTest.java
@@ -71,6 +71,18 @@ public class DockerBuildServiceTest {
   }
 
   @Test
+  public void build_withInvalidConfiguration_shouldNotBuildAndTag() throws Exception {
+    // When
+    image.setBuild(null);
+    new DockerBuildService(mockedJKubeServiceHub).build(image);
+    // Then
+    verify(mockedDockerBuildService, times(0))
+        .buildImage(eq(image), any(), any());
+    verify(mockedDockerBuildService, times(0))
+        .tagImage(anyString(), eq(image));
+  }
+
+  @Test
   public void build_withValidConfiguration_shouldBuildAndTag() throws Exception {
     // When
     new DockerBuildService(mockedJKubeServiceHub).build(image);

--- a/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildServiceTest.java
+++ b/jkube-kit/config/service/src/test/java/org/eclipse/jkube/kit/config/service/kubernetes/JibBuildServiceTest.java
@@ -213,6 +213,18 @@ public class JibBuildServiceTest {
     }
 
     @Test
+    public void build_withImageMissingBuildConfiguration_shouldNotBuildImage() throws JKubeServiceException {
+        // Given
+        imageConfiguration = ImageConfiguration.builder()
+            .name("test/foo:latest")
+            .build();
+        // When
+        new JibBuildService(mockedServiceHub).build(imageConfiguration);
+        // Then
+        jibServiceUtilMockedStatic.verify(() -> JibServiceUtil.buildContainer(any(), any(), any()), times(0));
+    }
+
+    @Test
     public void build_withImageBuildConfigurationSkipTrue_shouldNotBuildImage() throws JKubeServiceException {
         // Given
         imageConfiguration = ImageConfiguration.builder()

--- a/quickstarts/maven/ibm-javaee8-webprofile-liberty/pom.xml
+++ b/quickstarts/maven/ibm-javaee8-webprofile-liberty/pom.xml
@@ -49,6 +49,7 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <javaee.api.version>8.0</javaee.api.version>
         <jersey.version>2.26</jersey.version>
+        <junit.version>4.13.2</junit.version>
     </properties>
 
     <dependencyManagement>
@@ -62,12 +63,12 @@
             <dependency>
                 <groupId>javax</groupId>
                 <artifactId>javaee-api</artifactId>
-                <version>8.0</version>
+                <version>${javaee.api.version}</version>
             </dependency>
             <dependency>
                 <groupId>junit</groupId>
                 <artifactId>junit</artifactId>
-                <version>4.11</version>
+                <version>${junit.version}</version>
             </dependency>
             <dependency>
                 <groupId>org.glassfish</groupId>
@@ -108,12 +109,12 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-surefire-plugin</artifactId>
-                    <version>2.20</version>
+                    <version>2.22.2</version>
                 </plugin>
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-failsafe-plugin</artifactId>
-                    <version>2.20</version>
+                    <version>2.22.2</version>
                 </plugin>
                 <plugin>
                     <groupId>net.wasdev.wlp.maven.plugins</groupId>
@@ -123,7 +124,7 @@
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-war-plugin</artifactId>
-                    <version>3.2.0</version>
+                    <version>3.3.2</version>
                 </plugin>
             </plugins>
         </pluginManagement>

